GoRogue/MapGeneration/Region.cs

```diff
@@ -12,163 +14,117 @@ namespace GoRogue.MapGeneration
+            : this(corners, algorithm, components) { }
 
         /// <summary>
-        /// All of the boundary points of this region.
+        /// Returns a new Region with corners at the provided points.
         /// </summary>
-        public IReadOnlyArea OuterPoints => _outerPoints;
-        private MultiArea _outerPoints;
+        /// <param name="algorithm">Which Line-drawing algorithm to use</param>
+        /// <param name="corners">The points which are corners for this region</param>
+        public Region(Lines.Algorithm algorithm, params Point[] corners)
+            : this(corners, algorithm, null) { }
 
         /// <summary>
-        /// All of the points inside this region, excluding boundary points
+        /// Returns a new Region with corners at the provided points.
         /// </summary>
-        public IReadOnlyArea InnerPoints => _innerPoints;
-        private Area _innerPoints = null!;
+        /// <param name="components">A component collection to use for this region</param>
+        /// <param name="corners">The points which are corners for this region</param>
+        public Region(IComponentCollection? components, params Point[] corners)
+            : this(corners, Lines.Algorithm.DDA, components) { }
 
         /// <summary>
-        /// All of the outer points along the southern boundary
+        /// Returns a new Region with corners at the provided points.
         /// </summary>
-        public IReadOnlyArea SouthBoundary => _southBoundary;
-        private Area _southBoundary;
+        /// <param name="algorithm">Which Line-drawing algorithm to use</param>
+        /// <param name="components">A component collection for this region</param>
+        /// <param name="corners">The points which are corners for this region</param>
+        public Region(Lines.Algorithm algorithm, IComponentCollection? components, params Point[] corners)
+            : this(corners, algorithm, components) { }
 
         /// <summary>
-        /// All of the outer points along the northern boundary
+        /// Returns a new Region with corners at the provided points, using the algorithm DDA to produce lines
         /// </summary>
-        public IReadOnlyArea NorthBoundary => _northBoundary;
-        private Area _northBoundary;
+        /// <param name="corners">The corners of the region</param>
+        public Region(params Point[] corners) : this(corners, Lines.Algorithm.DDA, null) { }
 
-        /// <summary>
-        /// All of the outer points along the eastern boundary
-        /// </summary>
-        public IReadOnlyArea EastBoundary => _eastBoundary;
-        private Area _eastBoundary;
+        private Region(List<Point> corners, Lines.Algorithm algorithm, IComponentCollection? components)
+        {
+            Area = new PolygonArea(corners, algorithm);
+            GoRogueComponents = components ?? new ComponentCollection();
+            GoRogueComponents.ParentForAddedComponents = this;
+        }
```

Comment:
This set of constructors still doesn't quite work how we want.  Namely, note that we have the following implementation for the by-reference constructor:
```CS
public Region(ref List<Point> corners, Lines.Algorithm algorithm = Lines.Algorithm.DDA, IComponentCollection? components = null)
            : this(corners, algorithm, components) { }
```

This implementation is problematic because it results in the `corners` List being copied.  The call-chain evaluates to this:
- `this(corners, algorithm, components)`
    - Invokes `private Region(List<Point> corners, Lines.Algorithm algorithm, IComponentCollection? components) // Passing ref-list to non-ref list; This only copies the pointer so is technically fine, though perhaps not very clean`
        - Calls `Area = new PolygonArea(corners, algorithm); // Problem; non-ref List doesn't invoke the ref constructor of PolygonArea`
            - Calls `public PolygonArea(IEnumerable<Point> corners, Lines.Algorithm algorithm = Lines.Algorithm.DDA)`
                - Calls `this(corners.ToList(), algorithm) { } // Problem; ToList copies all elements of the list entirely`

One option that corrects the issue is to change the call to `PolygonArea` in the private constructor:
```CS
Area = new PolygonArea(ref corners, algorithm); // Added ref
```

This ensures that the only thing that is copied by `Region`'s ref constructor is the pointer, not the contents of the list.  It's not 100% clean but should be safe.